src/ctapipe/io/hdf5eventsource.py

```diff
@@ -622,58 +641,46 @@ def _generator(self):
                 if self.allowed_tels and tel_id not in self.allowed_tels:
                     continue
 
-                data.trigger.tel[tel_index.tel_id] = tel_trigger
-
-            if self.is_simulation:
-                data.simulation.shower = next(mc_shower_reader)
-
-            for kind, readers in dl2_readers.items():
-                c = getattr(data.dl2.stereo, kind)
-                for algorithm, reader in readers.items():
-                    c[algorithm] = next(reader)
-
-            # this needs to stay *after* reading the telescope trigger table
-            # and after reading all subarray event information, so that we don't
-            # go out of sync
-            if len(data.trigger.tels_with_trigger) == 0:
-                continue
-
-            self._fill_array_pointing(data)
-            self._fill_telescope_pointing(data, pointing_interpolator)
-
-            for tel_id in data.trigger.tel.keys():
                 key = f"tel_{tel_id:03d}"
 
-                if self.allowed_tels and tel_id not in self.allowed_tels:
-                    continue
+                containers = {
+                    "index": tel_index,
+                    "dl0": DL0TelescopeContainer(trigger=tel_trigger),
+                }
 
-                if key in true_impact_readers:
-                    data.simulation.tel[tel_id].impact = next(true_impact_readers[key])
+                containers["pointing"] = self._fill_telescope_pointing(
+                    tel_index.obs_id,
+                    tel_id,
+                    time=tel_trigger.time,
+                    tel_pointing_interpolator=pointing_interpolator,
+                )
 
                 if DataLevel.R1 in self.datalevels:
-                    data.r1.tel[tel_id] = next(waveform_readers[key])
+                    containers["r1"] = next(waveform_readers[key])
 
-                    # TODO: Delete if we drop support for datamodel versions < v6.0.0
```

Comment:
This comment states that the deletion of the following line implies a change in supported data model versions, is this true? In that case the changelog needs to mention it.

Reply:
This is a merge artifact I think... will fix.